test/hook.lua

```diff
@@ -1,6 +1,23 @@
+buf = ''
+files = {}
+
 -- recv hook, m is message
-function recv(m)
-    print(m)
+function recv(path, m)
+    local code = 0
+    local file = ''
+    buf = buf..m
+    if (string.sub(buf, -4) == '\r\n\r\n') then
+        local method = string.match(buf, "%S+")
```

Comment:
Maybe this would be simpler:
```
local name = buf:match('GET /(%S+) HTTP/%d%.%d\r\n')
```
You can assume that, if you can't match GET, the method is incorrect.

ps: untested code!
pps: Instead of `/` you should use `path's` value and, also, take a look at this: https://tools.ietf.org/html/rfc3986#section-3.3 to create a more precise pattern